File: codegen/swagger-ehr/dart/lib/model/admin_entry.dart
part of swagger.api;

class AdminEntry {
  
  CodePhrase language = null;

  CodePhrase encoding = null;

  List<Participation> otherParticipations = [];

  ObjectRef workflowId = null;

  PartyProxy subject = null;

  PartyProxy provider = null;

  String type = "ADMIN_ENTRY";

  ItemStructure data = null;

  AdminEntry();

  @override
  String toString() {
    return 'AdminEntry[language=$language, encoding=$encoding, otherParticipations=$otherParticipations, workflowId=$workflowId, subject=$subject, provider=$provider, type=$type, data=$data, ]';
  }

  AdminEntry.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    language = new CodePhrase.fromJson(json['language']);
    encoding = new CodePhrase.fromJson(json['encoding']);
    otherParticipations = Participation.listFromJson(json['other_participations']);
    workflowId = new ObjectRef.fromJson(json['workflow_id']);
    subject = new PartyProxy.fromJson(json['subject']);
    provider = new PartyProxy.fromJson(json['provider']);
    type = json['_type'];
    data = new ItemStructure.fromJson(json['data']);
  }

  Map<String, dynamic> toJson() {
    return {
      'language': language,
      'encoding': encoding,
      'other_participations': otherParticipations,
      'workflow_id': workflowId,
      'subject': subject,
      'provider': provider,
      '_type': type,
      'data': data
     };
  }

  static List<AdminEntry> listFromJson(List<dynamic> json) {
    return json == null ? new List<AdminEntry>() : json.map((value) => new AdminEntry.fromJson(value)).toList();
  }

  static Map<String, AdminEntry> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, AdminEntry>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new AdminEntry.fromJson(value));
    }
    return map;
  }
}

File: codegen/oas-ehr/spring/src/main/java/org/openapitools/model/AdminEntry.java
package org.openapitools.model;

import java.net.URI;
import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import org.openapitools.model.AbstractEntry;
import org.openapitools.model.ItemStructure;
import java.util.List;
import org.openapitools.jackson.nullable.JsonNullable;
import java.time.OffsetDateTime;
import javax.validation.Valid;
import javax.validation.constraints.*;
import io.swagger.v3.oas.annotations.media.Schema;


import java.util.*;
import javax.annotation.Generated;

/**
 * AdminEntry
 */


@Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
public class AdminEntry extends AbstractEntry {

  private String type = "ADMIN_ENTRY";

  private ItemStructure data;

  public AdminEntry() {
    super();
  }

  /**
   * Constructor with only required parameters
   */
  public AdminEntry(ItemStructure data) {
    super(language, encoding, subject);
    this.data = data;
  }

  public AdminEntry type(String type) {
    this.type = type;
    return this;
  }

  /**
   * Get type
   * @return type
  */
  
  @Schema(name = "_type", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("_type")
  public String getType() {
    return type;
  }

  public void setType(String type) {
    this.type = type;
  }

  public AdminEntry data(ItemStructure data) {
    this.data = data;
    return this;
  }

  /**
   * Get data
   * @return data
  */
  @NotNull @Valid 
  @Schema(name = "data", requiredMode = Schema.RequiredMode.REQUIRED)
  @JsonProperty("data")
  public ItemStructure getData() {
    return data;
  }

  public void setData(ItemStructure data) {
    this.data = data;
  }


  public AdminEntry language(CodePhrase language) {
    super.language(language);
    return this;
  }

  public AdminEntry encoding(CodePhrase encoding) {
    super.encoding(encoding);
    return this;
  }

  public AdminEntry otherParticipations(List<@Valid Participation> otherParticipations) {
    super.otherParticipations(otherParticipations);
    return this;
  }

  public AdminEntry addOtherParticipationsItem(Participation otherParticipationsItem) {
    super.addOtherParticipationsItem(otherParticipationsItem);
    return this;
  }

  public AdminEntry workflowId(ObjectRef workflowId) {
    super.workflowId(workflowId);
    return this;
  }

  public AdminEntry subject(PartyProxy subject) {
    super.subject(subject);
    return this;
  }

  public AdminEntry provider(PartyProxy provider) {
    super.provider(provider);
    return this;
  }

  public AdminEntry name(DvText name) {
    super.name(name);
    return this;
  }

  public AdminEntry archetypeNodeId(String archetypeNodeId) {
    super.archetypeNodeId(archetypeNodeId);
    return this;
  }

  public AdminEntry uid(UidBasedId uid) {
    super.uid(uid);
    return this;
  }

  public AdminEntry links(List<@Valid Link> links) {
    super.links(links);
    return this;
  }

  public AdminEntry addLinksItem(Link linksItem) {
    super.addLinksItem(linksItem);
    return this;
  }

  public AdminEntry archetypeDetails(Archetyped archetypeDetails) {
    super.archetypeDetails(archetypeDetails);
    return this;
  }

  public AdminEntry feederAudit(FeederAudit feederAudit) {
    super.feederAudit(feederAudit);
    return this;
  }
  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AdminEntry adminEntry = (AdminEntry) o;
    return Objects.equals(this.type, adminEntry.type) &&
        Objects.equals(this.data, adminEntry.data) &&
        super.equals(o);
  }

  @Override
  public int hashCode() {
    return Objects.hash(type, data, super.hashCode());
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class AdminEntry {\n");
    sb.append("    ").append(toIndentedString(super.toString())).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    data: ").append(toIndentedString(data)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

